homework/shared_ptr/tests.cpp

```diff
@@ -48,73 +58,177 @@ TEST_F(SharedPointerTest, shouldCopySharedPointer) {
 
     ASSERT_EQ(*s_ptr, *newPtr);
     ASSERT_EQ(*s_ptr, *newPtr1);
+    ASSERT_EQ(s_ptr.use_count(), threeReferences);
+
+    auto newRPtr(rs_ptr);
+    auto newRPtr1 = rs_ptr;
+
+    ASSERT_EQ(*rs_ptr, *newRPtr);
+    ASSERT_EQ(*rs_ptr, *newRPtr1);
+    ASSERT_EQ(rs_ptr.use_count(), threeReferences);
 }
 
 TEST_F(SharedPointerTest, shouldMoveSharedPointer) {
     auto newPtr(std::move(s_ptr));
     ASSERT_EQ(*newPtr, testValue);
+    ASSERT_EQ(newPtr.use_count(), oneReference);
+
+    auto newRPtr(std::move(rs_ptr));
+    ASSERT_EQ(*newRPtr, testValue);
+    ASSERT_EQ(newRPtr.use_count(), oneReference);
 }
 
-TEST_F(SharedPointerTest, shouldResetSharedPtr) {
-    s_ptr.reset();
+TEST_F(SharedPointerTest, shouldUseCopyAssingnmentOperatorSharedPointer) {
+    SharedPointer<int> newPtr;
```

Comment:
please use shared_ptr that already manages some memory.
```suggestion
    SharedPointer<int> newPtr{new int{}};
```